### PR TITLE
split genericapiserver configuration apart so that you can run withou…

### PR DESCRIPTION
**Pull request by deads2k**
…t flag options

@dims Take a look at this re-slicing of the `genericapiserver.Config` creation.   I think this helps composers overall and resolves the chicken/egg problem you were having.

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/34556)

### CODE REVIEW

**Comment by dims** [comment]
I've tested this with my PR https://github.com/kubernetes/kubernetes/pull/33966 and it works well. One worry about this PR is how it will affect the defaults. `--insecure-address=0` on command line was not honored in an earlier revision/commit. So would love others to take a look at this PR as well.

**Comment by dims** [comment]
I will wait for the builds to go green and do a full review as well.

**Comment by dims** [comment]
@k8s-bot unit test this : #33461

**Comment by dims** [comment]
@k8s-bot cvm gce e2e test this : #33388

**Comment by dims** [comment]
@k8s-bot bootstrap gce e2e test this : #IGNORE (logs missing)

**Comment by dims** [comment]
@k8s-bot bootstrap gce e2e test this : #IGNORE (logs missing)

File: pkg/genericapiserver/config.go [hunk 1]
```diff
@@ -223,57 +258,44 @@ func NewConfig(options *options.ServerRunOptions) *Config {
...
 type completedConfig struct {
 	*Config
 }
 
-// Complete fills in any fields not set that are required to have valid data. It's mutating the receiver.
+// Complete fills in any fields not set that are required to have valid data and can be derived
+// from other fields.  If you're going to `ApplyOptions`, do that first.  It's mutating the receiver.
 func (c *Config) Complete() completedConfig {
-	if c.ServiceClusterIPRange == nil {
+	if c.ServiceClusterIPRange.IP == nil {
```
Comment (dims) [inline pkg/genericapiserver/config.go]: Should we check both `c.ServiceClusterIPRange == nil && c.ServiceClusterIPRange.IP == nil` ?

Comment (deads2k) [inline pkg/genericapiserver/config.go]: > Should we check both c.ServiceClusterIPRange == nil && c.ServiceClusterIPRange.IP == nil ?

sure, will update.

Comment (dims) [inline pkg/genericapiserver/config.go]: @deads2k Nice LGTM 👍  once that change is in

**Comment by deads2k** [comment]
@k8s-bot cvm gce e2e test this: issue #34577

**Pull request merged**